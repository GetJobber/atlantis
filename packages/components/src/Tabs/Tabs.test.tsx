import React from "react";
import renderer from "react-test-renderer";
import { cleanup, fireEvent, render } from "@testing-library/react";
import { Tab, Tabs } from ".";

afterEach(cleanup);

const omelet = (
  <Tabs>
    <Tab label="Eggs">
      <p>🍳</p>
      <p>Eggs</p>
    </Tab>
    <Tab label="Cheese">
      <p>🧀</p>
    </Tab>
  </Tabs>
);

it("renders Tabs", () => {
  const tree = renderer.create(omelet).toJSON();
  expect(tree).toMatchSnapshot();
});

test("it should switch tabs", () => {
  const { getByText, queryByText } = render(omelet);

  expect(queryByText("🍳")).toBeTruthy();
  expect(queryByText("🧀")).toBeFalsy();

  fireEvent.click(getByText("Cheese"));
  expect(queryByText("🍳")).toBeFalsy();
  expect(queryByText("🧀")).toBeTruthy();

  fireEvent.click(getByText("Eggs"));
  expect(queryByText("🍳")).toBeTruthy();
  expect(queryByText("🧀")).toBeFalsy();
});
