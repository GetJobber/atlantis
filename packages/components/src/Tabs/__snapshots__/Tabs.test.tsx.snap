// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Tabs 1`] = `
<div
  className="tabs"
>
  <div
    className="overflow"
  >
    <ul
      className="tabRow"
      role="tablist"
    >
      <li
        role="presentation"
      >
        <button
          className="tab selected"
          id="Eggs"
          onClick={[Function]}
          role="tab"
          type="button"
        >
          <span
            className="base regular base green"
          >
            Eggs
          </span>
        </button>
        <button
          className="tab"
          id="Cheese"
          onClick={[Function]}
          role="tab"
          type="button"
        >
          <span
            className="base regular base heading"
          >
            Cheese
          </span>
        </button>
      </li>
    </ul>
  </div>
  <section
    aria-label="Eggs"
    className="tabContent"
    role="tabpanel"
  >
    <p>
      üç≥
    </p>
    <p>
      Eggs
    </p>
  </section>
</div>
`;
